cmd/lifecycle/main.go

```diff
@@ -19,17 +19,17 @@ func main() {
 
 	switch strings.TrimSuffix(filepath.Base(os.Args[0]), filepath.Ext(os.Args[0])) {
 	case "detector":
-		cmd.Run(&detectCmd{}, false)
+		cmd.Run(&detectCmd{platformAPI: platformAPI}, false)
 	case "analyzer":
-		cmd.Run(&analyzeCmd{}, false)
+		cmd.Run(&analyzeCmd{platformAPI: platformAPI}, false)
 	case "restorer":
-		cmd.Run(&restoreCmd{}, false)
+		cmd.Run(&restoreCmd{platformAPI: platformAPI}, false)
 	case "builder":
 		cmd.Run(&buildCmd{buildArgs: buildArgs{platformAPI: platformAPI}}, false)
 	case "exporter":
 		cmd.Run(&exportCmd{exportArgs: exportArgs{platformAPI: platformAPI}}, false)
 	case "rebaser":
-		cmd.Run(&rebaseCmd{}, false)
+		cmd.Run(&rebaseCmd{platformAPI: platformAPI}, false)
```

Comment:
When I added `platformAPI` to the commands, I was wondering whether I should add them as they're right now in the PR or I make them part of `*Args`. I thought that my idea is better because I didn't want to pass the value to functions that don't care about it (although it'll be consistent with `buildArgs` and `exportArgs`. But if you think that we'll need it in the **near** future, I'll make this change.

Reply:
I'm not sure I understand. Should I make any changes to the link you indicated in the `creator` (https://github.com/buildpacks/lifecycle/blob/main/cmd/lifecycle/creator.go#L167)?